### ISSUE TITLE
Performance issue with loading Stock page due to very slow StockRepository.php query

### ISSUE THREAD

**Issue by ced--ced** [PrestaShop/PrestaShop#37691]
### Prerequisites

- [x] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [x] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

When accessing the Sell > Catalog > Stock page, the product list takes 2 to 3 minutes to load, which sends to 100% the CPU use of the database server and slows down the whole website, back and front. The website has only 350 products with 1 or 2 attributes.

After several searches, I have found that commenting the following function in StockRepository.php in line 190 solves the issue and the list loads immediately.

`
/**
* @param QueryParamsCollection $queryParams
*  *
* @return mixed
*/
public function getData(QueryParamsCollection $queryParams)
    {
        $this->stockManager->updatePhysicalProductQuantity(
            $this->getContextualShopId(),
            $this->orderStates['error'],
            $this->orderStates['cancellation']
        );

        return parent::getData($queryParams);
    }
`

But modifying a core file is not the solution and I don't know if there can be consequences elsewhere where the function might also be called.

### Steps to reproduce

1. Go to Sell > Catalog > Stock page.
2. The list might not load or take 2-3 minutes to load.
3. Edit the file /src/PrestaShopBundle/Entity/Repository/StockRepository.php and comment the function updatePhysicalProductQuantity in line 190.
4. The list loads immediately.

### PrestaShop version(s) where the bug happened

8.2.0

### PHP version(s) where the bug happened

8.1

**Comment by paulnoelcholot** [comment]
Hello @ced--ced,

I try to reproduce your issue but as you can see, I can't : 
https://www.loom.com/share/78effe3a17b946fc92ad6e9b71cb4f2c?sid=5eac2474-e61c-4c85-aa53-1006b74f07af

Did I miss something?

Thanks for your feedback!

**Comment by kpodemski** [comment]
I can confirm this issue, please take a look at the PR below for a possible solution

https://github.com/PrestaShop/PrestaShop/pull/34863

**Comment by ced--ced** [comment]
@kpodemski We have implemented the solution but we now have numerous errors in the backoffice. Prestashop looks for tables with the prefix ps_ which is not the correct prefix in the database nor in the parameters.php file. Is it possible there's an error in the code provided in the fix #34863? Thanks

**Comment by kpodemski** [comment]
@ced--ced yes, there are some obvious errors, like lack of handling the prefix, you need to adjust the code a bit - that's why this fix hasn't been implemented into the core just yet

**Comment by ced--ced** [comment]
Ok thanks @kpodemski. We have a look at it.

**Comment by paulnoelcholot** [comment]
Hello @ced--ced and @kpodemski,

I tested with 1000+ products this time and that included images but I still have no latency issues. Could you tell me how to reproduce this issue please?

https://www.loom.com/share/f43b1f6e028e4486a0cfe75f896c0e71?sid=8d5d761d-3695-4727-ab87-30c6783b62cd

**Comment by kpodemski** [comment]
@paulnoelcholot you need to have a lot of orders that are neither canceled or processed already where stock manager has to synchronize reserved quantity for them

**Comment by ced--ced** [comment]
Indeed we have about 50 000 orders. We're testing some corrections to the code. I will let you know if it works better.

**Comment by santietsit** [comment]
It worked, thanks a lot!

**Comment by SiraDIOP** [comment]
Hello @kpodemski, 

I put the label verified on issue and  the PR will be tested by dev.

**Comment by boo-code** [comment]
The code named in the report is unchanged on 9.2, and the structural part of the problem is measurable, but I could not reproduce the two to three minutes on MySQL 8.4 and want to be precise about which is which.

`StockRepository::getData()` still recomputes the whole shop on every render, with no product restriction:

```php
public function getData(QueryParamsCollection $queryParams)
{
    $this->stockManager->updatePhysicalProductQuantity(
        $this->getContextualShopId(),
        $this->orderStates['error'],
        $this->orderStates['cancellation']
    );

    return parent::getData($queryParams);
}
```

`getDataExport()` calls the same method, so the export pages pay it too. Underneath, `updateReservedProductQuantity()` is an `UPDATE ps_stock_available` whose `SET` is a correlated subquery joining `orders`, `order_detail` and `order_state`, evaluated once per stock row of the shop. So a page that displays one page of rows writes to every row in the catalogue first.

Measured on a 9.2 shop, MySQL 8.4, 1058 `stock_available` rows and 90 016 `order_detail` rows spread over 1000 products:

```
whole shop recompute (what the page does today): 0.112 s
single product recompute:                        0.005 s
```

So the call is about twenty times the work a single page of the grid needs, and it grows with the catalogue while the page size does not. That is the part I can demonstrate.

What I cannot demonstrate is the reported magnitude. At that data volume MySQL 8.4 resolves the correlated subquery through the index on `order_detail`, and the whole call stays around a tenth of a second, which is consistent with @paulnoelcholot not reproducing it. Reaching minutes presumably needs a different optimiser or a data shape I did not hit, so the severity looks engine or data dependent rather than universal.

The shape of a fix that does not depend on any of that: scope the synchronisation to the products actually being displayed, since the grid reads back what it just wrote. Reading the page, syncing that set of products, then reading it again is bounded by the page size instead of the catalogue. It does change one thing worth deciding deliberately, that visiting the Stock page would no longer repair drift on products outside the current page, so it is worth confirming that is acceptable before it is written.
